### ISSUE TITLE
[Enhancement] Add AsciiDoc support

### ISSUE THREAD

**Issue by r0ckarong** [doxygen/doxygen#8506]
It would be great if doxygen supported the AsciiDoc format for writing additional documentation and comments. The markdown support was added 9 years ago but the "language" has not evolved in a sensible way despite it's growing use and rather fragmented into competing subsets and implementations. AsciiDoc, on the other hand, is currently being standardized by the Eclipse foundation and the ecosystem has greatly improved over the last years.

In my dream world, doxygen would provide the means to write comments and documentation in AsciiDoc format and provide a means of accessing doxygen output from the AsciiDoc ecosystem for building other documentation artifacts (most specifically PDF).

**Comment by gdolle** [comment]
+1
I agree it would be really great to have an asciidoc output generator.
I think this would perfectly fit with [antora](https://antora.org) to generate an API stored on a repository.

**Comment by r0ckarong** [comment]
> I think this would perfectly fit with antora to generate an API stored on a repository.
I am also thinking about Antora for this. Some conversion that allows users to generate Antora output and configure it to produce a complete documentation.

**Comment by arwedus** [comment]
Doxygen markdown support basically converts all markdown syntax to the corresponding Doxygen commands. This works because Doxygen provides an equivalent for all Markdown mark-up commands. Not sure the same would hold true for AsciiDoc.

Using the Doxygen XML output for import in a different documentation generator is actually quite a standard way to achieve integration of code documentation in the documentation generator of your choice.  E.g. Sphinx supports this via sphinx-breathe.

**Comment by gdolle** [comment]
> Doxygen provides an equivalent for all Markdown mark-up commands
Yes, that's what should be implemented but for basic asciidoc usage.

> Using the Doxygen XML output for import in a different documentation generator is actually quite a standard way
Agree, but the issue is to extend existing doxygen input/[output format](https://www.doxygen.nl/manual/output.html) and avoid XML intermediates as mentioned @r0ckarong in the last paragraph (note that some people use rst/mardown/asciidoc output directly, without any doc gen).

**Comment by JakeSmarter** [comment]
Yes, it would be so great to have a single formally complete documentation lightweight markup language. Does it have to be either AsciiDoc or CommonMark? I do not care much as long as it is consistent across the source code and non-source code domains and the tools produce high quality publishing formats like PDF or EPUB. It seems though that AsciiDoc has at least set out to achieve this goal. [Quarto](https://quarto.org) seems to be the Markdown contender in this regard but has a deep dependency chain, which may be better or worse.
